### DISCUSSION TITLE
Read tabix indexed BED file?

### DISCUSSION THREAD

**Discussion by cademirch** [zaeleus/noodles#312] [Q&A] [answered]
Quick question that I'm probably missing the obvious answer but am in too deep to see it. Any chance for an example on how to read a tabix indexed bed file? Its not clear to me how to achieve this given the examples for both the tabix reader and bed reader. 

Thanks!

**Comment by zaeleus** [comment, marked as answer]
The BED reader currently doesn't have a convenient method to query in the input. However, you can compose your own query ([`csi::io::Query`](https://docs.rs/noodles/0.85.0/noodles/csi/io/struct.Query.html)) with a BED reader ([`bed::io::Reader`](https://docs.rs/noodles/0.85.0/noodles/bed/io/struct.Reader.html)) using existing componenets as a workaround. I added an example of this at [`noodles-bed/examples/bed_query.rs`](https://github.com/zaeleus/noodles/blob/d2b2cee1a94bdd0b3f7479102d8ee76a85e56fcf/noodles-bed/examples/bed_query.rs).

**Reply by cademirch** [reply]
Awesome. Thank you so much @zaeleus! This is very helpful - both for my project, and learning Rust.
